src/Components/GameBoard.js

```diff
@@ -419,40 +431,94 @@ class GameBoard extends Component {
 
   /* Define custom graph editing methods here */
 
+  //checks if the current board is balanced and returns true or false
+  checkRebalance = () => {
+    let isBalanced = this.state.board.graph.balanced
+    console.log("balanced: ",this.state.board.graph.balanced)
+    return isBalanced
+
+  }
+  //called if checkRebalance returns false
+  //post request to get correct/balanced game board and sets gameboard to 
+  //return balanced board
+  rebalance = async () =>{
```

Comment:
We plan to do the functional rebalance instead of the automatic mock rebalance call, explained by @kulriz.